R/terms.R

```diff
@@ -170,3 +169,234 @@ get_term_label <- function(term_iris, preserveOrder = FALSE, verbose = FALSE) {
 
   res
 }
+
+#' Terminfo Objects
+#'
+#' Create and test objects of type "terminfo", and extract properties from them.
+#'
+#' @description
+#' `as.terminfo` creates an object (or a list of objects) of type "terminfo".
+#'   The object to be coerced can be a character vector (of term IRIs), or a data.frame.
+#'   In the latter case, there must be a column "id" with term IRIs.
+#'   If the object is already of type "terminfo", a new terminfo object will be created
+#'   so options such as withClassification will be applied.
+#'
+#' @param x an object of type "terminfo" or coercible to it, or to be tested
+#'  for being of type "terminfo"
+#' @param withClassification logical. If TRUE classification data will be available
+#'  through the terminfo object at key "classification". Default is FALSE, because
+#'  obtaining taxa requires an additional query per object. The default can be
+#'  customized by setting the "rphenoscape.fetch.classification" option.
+#' @param ... additional parameters where applicable
+#'
+#' @return
+#' `as.terminfo` returns and object of type "terminfo", or a list of such objects
+#'    if the object to be coerced had multiple elements (if a vector) or rows (if a
+#'    data.frame). A terminfo object has properties "id" (ID, i.e., IRI of the
+#'    term), "label" (label of the term if one exists). If the term is a taxon 
+#'    additional properties will be populated: "extinct", "rank", "common_name".
+#'    If `withClassification` is TRUE,there will also be a key "classification"
+#'    (a list with properties "subClassOf", "equivalentTo" and "superClassOf").
+#'
+#' @examples
+#' # find a term iri
+#' term_iri <- find_term('basihyal bone', matchType='exact')
+#' # turn it into a terminfo object
+#' obj <- as.terminfo(term_iri)
+#' class(obj)
+#' obj
+#'
+#' # find another term iri
```

Comment:
```suggestion
#' # taxon terms have additional properties: 
```

Nitpicking here, but I think it's useful as this isn't just some other term IRI.